### PR TITLE
[Bugfix] Fix target host for vm compiler

### PR DESCRIPTION
**Pull request by icemelon9**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

cc @wweic

### CODE REVIEW

File: python/tvm/relay/backend/vm.py [hunk 1]
```diff
@@ -178,6 +178,11 @@ def compile(self, mod, target=None, target_host=None):
         """
         target = _update_target(target)
         target_host = None if target_host == "" else target_host
+        if not target_host:
```
Comment (yongwww) [inline python/tvm/relay/backend/vm.py]: duplicate if condition with line 182

Comment (icemelon9) [reply python/tvm/relay/backend/vm.py]: After this if block, it's still possible that `target_host` is None. So it's necessary to keep the if condition in line 186

**Comment by zhiics** [comment]
We can test once heterogeneous execution is added. Or we can still pass a dict to VM, but it seems not to be good test anyway.

**Comment by icemelon9** [comment]
Yes, it's not easy to test this unless we have a performance test.

**Pull request merged**